vscode/src/repository/githubRepoMetadata.ts

```diff
@@ -10,24 +10,54 @@ import {
 } from '@sourcegraph/cody-shared'
 import { Observable, map } from 'observable-fns'
 import { logDebug } from '../output-channel-logger'
+import { localStorage } from '../services/LocalStorageProvider'
 import { remoteReposForAllWorkspaceFolders } from './remoteRepos'
 
+export interface RepoAccessibilityData {
+    repoName: string
+    isPublic: boolean
+}
+
 interface GitHubDotComRepoMetaData {
     // The full uniquely identifying name on github.com, e.g., "github.com/sourcegraph/cody"
     repoName: string
     isPublic: boolean
 }
 
+const DEFAULT_MIN_LOCAL_STORAGE_UPDATE_TIME_MS = 1000 * 60 * 10 // 10 minutes
+
 export class GitHubDotComRepoMetadata {
     // This class is used to get the metadata from the gitApi.
     private static instance: GitHubDotComRepoMetadata | null = null
+    // Store a copy of the latest local storage data for comparison with the current cache.
+    private latestLocalStorageData: RepoAccessibilityData[] = []
+    // Last time when the local storage was updated.
+    private lastLocalStorageUpdateTime: number | null = null
+    // Since the local storage update can be expansive, we add a minimum time between updates.
+    private readonly minLocalStorageUpdateTimeMs: number
+
     private cache = new Map<string /* repoName */, GitHubDotComRepoMetaData | undefined>()
 
-    private constructor() {}
+    private constructor(minLocalStorageUpdateTimeMs: number) {
+        this.minLocalStorageUpdateTimeMs = minLocalStorageUpdateTimeMs
+        this.populateCacheFromLocalStorage()
+    }
+
+    public populateCacheFromLocalStorage(): void {
+        this.cache.clear()
+        this.latestLocalStorageData = localStorage.getGitHubRepoAccessibility()
+        for (const data of this.latestLocalStorageData) {
+            this.cache.set(data.repoName, { repoName: data.repoName, isPublic: data.isPublic })
```

Comment:
Could it be reduced to this?

```suggestion
            this.cache.set(data.repoName, data)
```